lio_sam:

  # voxel filter paprams
  odometrySurfLeafSize: 0.2   #0.2                          # default: 0.4 - outdoor, 0.2 - indoor  
  mappingCornerLeafSize: 0.1  #0.1                   # default: 0.2 - outdoor, 0.1 - indoor 
  mappingSurfLeafSize: 0.2    #0.2                   # default: 0.4 - outdoor, 0.2 - indoor   
  mappingFullLeafSize: 0.05    #0.01                  # default: 0.01  save map

  # robot motion constraint (in case you are using a 2D robot)
  z_tollerance: 50                              # meters  1000
  rotation_tollerance: 100                      # radians  1000

  # CPU Params
  numberOfCores: 8                              # number of cores for mapping optimization
  mappingProcessInterval: 0.1  #0.2             # seconds, regulate mapping frequency ,default: 0.15

  # Surrounding map
  surroundingkeyframeAddingDistThreshold: 0.5         # meters, regulate keyframe adding threshold
  surroundingkeyframeAddingAngleThreshold: 0.2        # radians, regulate keyframe adding threshold  0.2
  surroundingKeyframeDensity: 1.0        # meters, downsample surrounding keyframe poses  2.0   0.5
  surroundingKeyframeSearchRadius: 50.0               # meters, within n meters scan-to-map optimization (when loop closure disabled); default: 50

  # Loop closure
  loopClosureEnableFlag: false
  loopClosureFrequency: 1.0                     # Hz, regulate loop closure constraint add frequency default: 1.0
  surroundingKeyframeSize: 20.0                 # submap size (when loop closure enabled) default: 50.0   20.0
  historyKeyframeSearchRadius: 10.0 #35.0       # meters, key frame that is within n meters from current pose will be considerd for loop closure default: 15.0
  historyKeyframeSearchTimeDiff: 60.0 #40.0     # seconds, key frame that is n seconds older will be considered for loop closure default: 30
  historyKeyframeSearchNum: 20                  # number of hostory key frames will be fused into a submap for loop closure default: 25
  historyKeyframeFitnessScore: 0.15              # icp threshold, the smaller the better alignment 0.3

  # Visualization
  globalMapVisualizationSearchRadius: 100.0     # meters, global map visualization radius
  globalMapVisualizationPoseDensity: 10.0       # meters, global map visualization keyframe density
  globalMapVisualizationLeafSize: 0.1 #1.0      # meters, global map visualization cloud density

  # visual iktree_map   
  visulize_IkdtreeMap: false

  # reconstruct iktree_map  
  recontructKdTree: false
  angleThreshold: 0.1 # angleThreshold to judge reconstruct or not
  transThreshold: 0.1 # transThreshold to judge reconstruct or not
  # back-end
  backEndEnable: true
  # update lio state or not 
  updateLio: true
  isNeedLocalMap: false
  isFullMap: true

  # Export settings
  saveFrameInterval: 15
  savePCD: false                        # https://github.com/TixiaoShan/LIO-SAM/issues/3
  savePCDDirectory: "/home/Hope/SLAM/map_save/fastlio_map/" #"/home/haige750/map_save/fastlio_map/" #

  carRadius: 0.43

gps:
  bUseGps: true # use gps or not 
  debugGps: false # debug
  addGps: false # add gps factor or not
  gpsInitOrientation: true # use gps to init orientation
  useGpsElevation: true #use gps Elevation or not 
  rpyInitMode: 2 #0:imu, 1:beidou, 2:set use to init yaw
  bUseGnssYaw: true

  gpsTopic: "/beidou" #"/gps/fix" #
  rpyTopic: "/bd_rpy"
  imuTopic: "/imu_raw"
  gpsAltitudeNoiseScore: 10.0
  gpsLatitudeNoiseScore: 50.0
  gpsFrequence: 10 # Frequence of gps
  gpsInitPointCnt: 20 # number of gps, first add
  gpsCovThreshold: 2.0 # covariance of gps to add
  gpsNoiseDefault: 0.5 #0.02
  gpsNoiseScale: 50
  gpsDistance: 0.5 # two gps distance to add a gps factor
  RPYGPS2Imu: [0, 0, 0] # rotation from gps to imu [0, 0, -0.012]